shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java

```diff
@@ -125,6 +125,83 @@ public static void writeCSVFile(Collection<? extends PropertyKey> defaultKeys, S
   }
 
   /**
+   * Writes description of property key to yml files.
+   *
+   * @param defaultKeys Collection which is from PropertyKey DEFAULT_KEYS_MAP.values()
+   * @param filePath    path for csv files
+   */
+  public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, String filePath)
+      throws IOException {
+    if (defaultKeys.size() == 0) {
+      return;
+    }
+
+    FileWriter fileWriter;
+
+    //HashMap for yml file names
+    Map<String, String> fileNames = new HashMap<>();
+    fileNames.put("user", PathUtils.concatPath(filePath, "user-configuration.yml"));
+    fileNames.put("master", PathUtils.concatPath(filePath, "master-configuration.yml"));
+    fileNames.put("worker", PathUtils.concatPath(filePath, "worker-configuration.yml"));
+    fileNames.put("security", PathUtils.concatPath(filePath, "security-configuration.yml"));
+    fileNames.put("keyvalue", PathUtils.concatPath(filePath, "key-value-configuration.yml"));
+    fileNames.put("common", PathUtils.concatPath(filePath, "common-configuration.yml"));
+
+    //HashMap for FileWriter per each category
+    Map<String, FileWriter> fileWriterMap = new HashMap<>();
+    Closer closer = Closer.create();
+    try {
+      for (Map.Entry<String, String> entry : fileNames.entrySet()) {
+        fileWriter = new FileWriter(entry.getValue());
+        //put fileWriter
+        fileWriterMap.put(entry.getKey(), fileWriter);
+        //register file writer
+        closer.register(fileWriter);
+      }
+
+      //Sort defaultKeys
+      Comparator<PropertyKey> pC = new PropertyKeyComparator();
+      List<PropertyKey> dfkeys = new ArrayList<>(defaultKeys);
+      Collections.sort(dfkeys, pC);
+
+      for (PropertyKey iteratorPK : dfkeys) {
+        String pKey = iteratorPK.toString();
+        PropertyKey pk = PropertyKey.fromString(pKey);
+
+        String description = pk.getDescription();
+
+        //Persist property key and default value to yml files
```

Comment:
NIT: space after `//`, and I think I prefer "Write ..." instead of "Persist ..."